File: 0001-two-sum/0001-two-sum.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        numMap = {}

        # for each num 
        for i in range(len(nums)):
            num = nums[i]
            complement = target - num

            if num in numMap:
                return [i, numMap[num]]
            
            numMap[complement] = i

        return []

File: 0001-two-sum/0001-two-sum.js
/**
 * @param {number[]} nums
 * @param {number} target
 * @return {number[]}
 */
var twoSum = function(nums, target) {
    const numMap = {}

    for (let i = 0; i < nums.length; i++) {
        const comp = target - nums[i]
    
        // number exists
        if (nums[i] in numMap) {
            return [i, numMap[nums[i]]]
        } 
        // add to map comp : i
        else {
            numMap[comp] = i
        }
    }
    return []
    
};